L1Trigger/L1THGCal/src/HGCalTriggerCellCalibration.cc

```diff
@@ -44,25 +60,52 @@ void HGCalTriggerCellCalibration::calibrateInMipT(l1t::HGCalTriggerCell& trgCell
 
 void HGCalTriggerCellCalibration::calibrateMipTinGeV(l1t::HGCalTriggerCell& trgCell) const {
   const double MevToGeV(0.001);
+  double trgCellEt(0.);
 
-  DetId trgdetid(trgCell.detId());
-  unsigned trgCellLayer = triggerTools_.layerWithOffset(trgdetid);
+  unsigned trgCellLayer = triggerTools_.layerWithOffset(trgCell.detId());
 
-  if (dEdX_weights_.at(trgCellLayer) == 0.) {
-    throw cms::Exception("BadConfiguration")
-        << "Trigger cell energy forced to 0 by calibration coefficients.\n"
-        << "The configuration should be changed. "
-        << "Discarded layers should be defined in hgcalTriggerGeometryESProducer.TriggerGeometry.DisconnectedLayers "
-           "and not with calibration coefficients = 0\n";
-  }
+  if (DetId(trgCell.detId()).det() == DetId::HGCalTrigger and
+      HGCalTriggerDetId(trgCell.detId()).subdet() == HGCalTriggerSubdetector::HFNoseTrigger) {
+    if (dEdX_weights_Nose_.at(trgCellLayer) == 0.) {
+      throw cms::Exception("BadConfiguration - HFNose")
+          << "Trigger cell energy forced to 0 by calibration coefficients.\n"
+          << "The configuration should be changed. "
+          << "Discarded layers should be defined in "
+             "hgcalTriggerGeometryESProducer.TriggerGeometry."
+             "DisconnectedLayers "
+             "and not with calibration coefficients = 0\n";
+    }
+
+    /* weight the amplitude by the absorber coefficient in MeV/mip + bring it in
+     * GeV */
+    trgCellEt = trgCell.mipPt() * dEdX_weights_Nose_.at(trgCellLayer) * MevToGeV;
 
-  /* weight the amplitude by the absorber coefficient in MeV/mip + bring it in GeV */
-  double trgCellEt = trgCell.mipPt() * dEdX_weights_.at(trgCellLayer) * MevToGeV;
+    /* correct for the cell-thickness */
+    if (triggerTools_.isSilicon(trgCell.detId()) && thickCorr_ > 0) {
+      trgCellEt /= thickCorrNose_;
+    }
 
-  /* correct for the cell-thickness */
-  if (triggerTools_.isSilicon(trgdetid) && thickCorr_ > 0) {
-    trgCellEt /= thickCorr_;
+  } else {
+    if (dEdX_weights_.at(trgCellLayer) == 0.) {
+      throw cms::Exception("BadConfiguration - HGCAL ")
+          << "Trigger cell energy forced to 0 by calibration coefficients.\n"
+          << "The configuration should be changed. "
+          << "Discarded layers should be defined in "
+             "hgcalTriggerGeometryESProducer.TriggerGeometry."
+             "DisconnectedLayers "
+             "and not with calibration coefficients = 0\n";
+    }
+
+    /* weight the amplitude by the absorber coefficient in MeV/mip + bring it in
+     * GeV */
+    trgCellEt = trgCell.mipPt() * dEdX_weights_.at(trgCellLayer) * MevToGeV;
+
+    /* correct for the cell-thickness */
+    if (triggerTools_.isSilicon(trgCell.detId()) && thickCorr_ > 0) {
+      trgCellEt /= thickCorr_;
+    }
```

Comment:
There is unnecessary code duplication. Things could be grouped; only the weights and the thickness corrections need to be different.

Reply:
done